File: app/src/main/java/com/meta4projects/hilarityjokes/room/JokeDatabase.java
package com.meta4projects.hilarityjokes.room;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = Joke.class, version = 1, exportSchema = false)
public abstract class JokeDatabase extends RoomDatabase {

    private static volatile JokeDatabase INSTANCE;

    public static JokeDatabase getINSTANCE(final Context context) {
        if (INSTANCE == null) {
            synchronized (JokeDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = getDataBase(context);
                }
            }
        }

        return INSTANCE;
    }

    private static JokeDatabase getDataBase(final Context context) {
        return Room.databaseBuilder(context.getApplicationContext(),
                JokeDatabase.class, "joke_database").build();
    }

    public abstract JokeDao jokeDao();
}

File: app/src/main/java/com/meta4projects/hilarityjokes/room/JokeDatabase.kt
package com.meta4projects.hilarityjokes.room

import android.content.Context
import androidx.room.Database
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase

@Database(entities = [Joke::class], version = 2, exportSchema = true)
abstract class JokeDatabase : RoomDatabase() {
    abstract fun jokeDao(): JokeDao

    companion object {
        @Volatile
        private var INSTANCE: JokeDatabase? = null
        fun getINSTANCE(context: Context): JokeDatabase {
            return INSTANCE ?: synchronized(JokeDatabase::class.java) {
                val instance = getDataBase(context)
                INSTANCE = instance
                instance
            }
        }
    }
}

private fun getDataBase(context: Context): JokeDatabase {
    return databaseBuilder(context.applicationContext, JokeDatabase::class.java, "joke_database").build()
}
